lib/agent.js

```diff
@@ -1,9 +1,13 @@
 const tcp = require('bare-tcp')
 const HTTPClientConnection = require('./client-connection')
+const { EventEmitter } = require('bare-events')
 
-module.exports = class HTTPAgent {
+module.exports = class HTTPAgent extends EventEmitter {
   constructor(opts = {}) {
+    super()
+
     const { keepAlive = false, keepAliveMsecs = 1000 } = opts
+    this.options = opts
```

Comment:
```suggestion
    this.options = { keepAlive, keepAliveMsecs }
```

To avoid maintaining a reference to the caller-owned `opts`.